wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -257,10 +328,197 @@ private void returnConnectionBackToPool(StatefulRedisConnection <byte[], byte[]>
+    }
+  }
+
+  private void setupTraditionalAuthentication(Properties props) {
+    String username = CACHE_USERNAME.getString(props);
+    String password = CACHE_PASSWORD.getString(props);
+    props.setProperty("cacheUsername", username);
+    props.setProperty("cachePassword", password);
+  }
+
+  private String extractHostnameFromCacheEndpoint(Properties props) {
+    String endpoint = CACHE_RW_ENDPOINT_ADDR.getString(props);
+    if (endpoint != null && endpoint.contains(":")) {
+      return endpoint.split(":")[0];
+    }
+    return null;
+  }
+
+  private boolean isElastiCacheEndpoint(String hostname) {
+    return hostname != null && hostname.endsWith(".cache.amazonaws.com");
+  }
+
+  private boolean hasTraditionalCredentials(Properties props) {
+    String username = CACHE_USERNAME.getString(props);
+    String password = CACHE_PASSWORD.getString(props);
+    return username != null && password != null;
+  }
+
+  private void scheduleReAuthentication() {
+    if (!this.iamAuthEnabled) {
+      return;
+    }
+
+    if (reAuthScheduler == null) {
+      synchronized (this) {
+        if (reAuthScheduler == null) {
+          reAuthScheduler = Executors.newSingleThreadScheduledExecutor(r -> {
+            Thread t = new Thread(r, "ElastiCache-ReAuth");
+            t.setDaemon(true);
+            return t;
+          });
+        }
+      }
+    }
+
+    // Cancel existing task
+    if (reAuthTask != null) {
+      reAuthTask.cancel(false);
+    }
+
+    // Schedule re-auth at 11h50m
+    reAuthTask = reAuthScheduler.scheduleAtFixedRate(
+        this::performReAuthentication,
+        REAUTH_TRIGGER_MINUTES,
+        REAUTH_INTERVAL_HOURS * 60, // Convert hours to minutes
+        TimeUnit.MINUTES
+    );
+
+    LOGGER.info("Scheduled ElastiCache re-authentication every " + REAUTH_INTERVAL_HOURS + " hours");
+  }
+
+  private void performReAuthentication() {
+    try {
+      LOGGER.finest("[" + new java.util.Date() + "] Performing ElastiCache re-authentication...");
+
+      // Force token refresh by clearing cache
+      String username = CACHE_IAM_USERNAME.getString(connectionProperties);
+      String region = CACHE_IAM_REGION.getString(connectionProperties);
+      String hostname = this.hostAndPort[0];
+      int port = this.hostAndPort.length > 1 ? Integer.parseInt(this.hostAndPort[1]) : 6379;
+
+      String cacheKey = IamAuthUtils.getCacheKey(username, hostname, port, Region.of(region));
+      IamAuthCacheHolder.cacheRemoveToken(cacheKey);
+
+      // Invalidate connection pools to force recreation with new token
+      invalidateConnectionPools();
+
+      LOGGER.finest("ElastiCache re-authentication completed successfully");
+    } catch (Exception e) {
+      LOGGER.warning("ElastiCache re-authentication failed: " + e.getMessage());
+    }
+  }
+
+  private void invalidateConnectionPools() {
+    READ_LOCK.lock();
+    try {
+      if (readConnectionPool != null) {
+        readConnectionPool.close();
```

Comment:
Don't close the connection pool. Please keep the current connections open in the pool, just send an `AUTH <username> <password>` command to the server to re-auth.